packages/graphql/tests/integration/directives/cypher/cypher-auth.int.test.ts

```diff
@@ -0,0 +1,106 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createBearerToken } from "../../../utils/create-bearer-token";
+import type { UniqueType } from "../../../utils/graphql-types";
+import { TestHelper } from "../../../utils/tests-helper";
+
+describe("https://github.com/neo4j/graphql/issues/5270", () => {
+    let User: UniqueType;
+    let UserBlockedUser: UniqueType;
+    let Thing: UniqueType;
+
+    const secret = "secret";
+    const testHelper = new TestHelper();
+
+    beforeEach(async () => {
+        User = testHelper.createUniqueType("User");
+        UserBlockedUser = testHelper.createUniqueType("UserBlockedUser");
+        Thing = testHelper.createUniqueType("Thing");
+
+        const typeDefs = /* GraphQL */ `
+            type ${User} @node(labels: ["${User}"]) @authorization(
+                filter: [
+                    { where: { node: { NOT: { blockedUsers_SOME: { to: { id: "$jwt.sub" } } } } } },
+                ]
+            ) {
+                id: ID! @unique @id
+                blockedUsers: [${UserBlockedUser}!]! @relationship(type: "HAS_BLOCKED", direction: OUT)
+            }
+        
+            type ${UserBlockedUser} @node(labels: ["${UserBlockedUser}"]) @authorization(
+                filter: [
+                    { where: { node: { from: { id: "$jwt.sub" } } } }
+                ]
+            ) {
+                id: ID! @id @unique
+                from: ${User}! @relationship(type: "HAS_BLOCKED", direction: IN) @settable(onCreate: true, onUpdate: false)
+                to: ${User}! @relationship(type: "IS_BLOCKING", direction: OUT) @settable(onCreate: true, onUpdate: false)
+            }
+
+            type ${Thing} {
+                user: ${User} @cypher(statement: "OPTIONAL MATCH (u:${User} {id: $jwt.sub}) RETURN u", columnName: "u")
+            }
+        
+
+        `;
+        await testHelper.initNeo4jGraphQL({
+            typeDefs,
+            features: {
+                authorization: {
+                    key: secret,
+                },
+            },
+        });
+    });
+
+    afterEach(async () => {
+        await testHelper.close();
+    });
+
+    test("should nested users with cypher directive filtered according to authorization rule", async () => {
+        const query = `
+            query GetMe {
+                ${Thing.plural} {
+                    user {
+                        id
+                        __typename
+                    }
+                }
+            }
+        `;
+
+        const userId = "my-user-id";
+
+        await testHelper.executeCypher(`
+                CREATE (:${User} {id: "${userId}"})
```

Comment:
To validate the assertion below, we have to have at least another User, if not, the assertion will be good with or without the filter from the authorization directive.

Reply:
:thinking: the assertion is not 100% correct, as the bug fixed here is actually invalid cypher that doesn't even run, I'll take a look to improve the test